x/stakeibc/keeper/msg_server_rebalance_validators.go

```diff
@@ -26,15 +26,14 @@ func (k msgServer) RebalanceValidators(goCtx context.Context, msg *types.MsgReba
 		k.Logger(ctx).Error(fmt.Sprintf("Host Zone not found %s", msg.HostZone))
 		return nil, types.ErrInvalidHostZone
 	}
-	maxNumRebalance := cast.ToInt(msg.NumRebalance)
-	if maxNumRebalance < 1 {
-		k.Logger(ctx).Error(fmt.Sprintf("Invalid number of validators to rebalance %d", maxNumRebalance))
-		return nil, types.ErrInvalidNumValidator
-	}
-	if maxNumRebalance > 4 {
-		k.Logger(ctx).Error(fmt.Sprintf("Invalid number of validators to rebalance %d", maxNumRebalance))
-		return nil, types.ErrInvalidNumValidator
+
+	params := k.GetParams(ctx)
+	if (msg.NumRebalance < params.MinNumRebalance) || (msg.NumRebalance > params.MaxNumRebalance) {
+		return nil, fmt.Errorf("invalid number of validators to rebalance (%d)", msg.NumRebalance)
 	}
+	
+	maxNumRebalance := cast.ToInt(msg.NumRebalance)
```

Comment:
You can remove this line and change the forloop below (where maxNumRebalance is used) to 

```
for i := uint64(1); i <= maxNumRebalance; i++ {
```